job-service/src/main/java/com/hpe/caf/services/job/api/DatabaseHelper.java

```diff
@@ -462,6 +433,41 @@ public void cancelJob(final String partitionId, String jobId) throws Exception {
         }
     }
 
+    /**
+     * Cancels the specified jobs
+     */
+    public int cancelJobs(final String partitionId, String jobIdStartsWith, final List<String> labels, final String filter)
+            throws Exception {
+        int successfulCancellations = 0;
+        final int limit = 100;
+        try (
+                final Connection conn = DatabaseConnectionProvider.getConnection(appConfig);
+                final CallableStatement stmt = conn.prepareCall("{call cancel_jobs(?,?,?,?,?,?,?,?,?,?)}")
+        ) {
+            do {
+                final SortField sortField = JobSortField.CREATE_DATE;
```

Comment:
I'm not sure buildGetJobsStatement is helping, it's introduced a memory leak and we don't need to pass sort field etc